src/wallet/db.cpp

```diff
@@ -408,8 +408,8 @@ bool BerkeleyBatch::VerifyEnvironment(const fs::path& file_path, std::string& er
     LogPrintf("Using wallet %s\n", file_path.string());
 
     // Wallet file must be a plain filename without a directory
-    if (walletFile != fs::basename(walletFile) + fs::extension(walletFile))
-    {
+    fs::path wallet_file_path(walletFile);
+    if (walletFile != wallet_file_path.stem().string() + wallet_file_path.extension().string()) {
```

Comment:
Should just drop this check. The way `SplitWalletPath` is defined this can never be false:

https://github.com/bitcoin/bitcoin/blob/8da1aa471eb9f062c99c3134d6151961619e9da5/src/wallet/db.cpp#L66

Also, the way `stem()` and `extension()` are defined you could replace them with a call to `filename()`

```c++
if (walletFile != wallet_file_path.filename().string()) {
```

https://www.boost.org/doc/libs/1_69_0/libs/filesystem/doc/reference.html#path-decomposition